### DISCUSSION TITLE
RSS plugin not rendering relative links correctly

### DISCUSSION THREAD

**Discussion by brabster** [squidfunk/mkdocs-material#7007] [Q&A]
I've set up an RSS feed for my blog but it seems to be misbehaving, specifically rendering internal links to other pages and images as literally what I typed in my markdown, not what they end up as after build.

https://tempered.works/feed_rss_created.xml

The links in the most recent post are wrong:

`src="./assets/hero.webp"></p><p>When I wrote about <a href="../2024-03-23-irresponsible-expertise-install-python-package/index.md">dangerous advice`

Image should be: https://tempered.works/2024-03-31-exploring-setup-py/assets/hero.webp
Link should be: https://tempered.works/posts/2024/03/23/irresponsible-expertise---python-packages/

Guessing I've not missed something as the same issue seems to be happening on mkdocs-material's own blog but maybe? Did I miss something or is this an issue?

Here's my settings:
![image](https://github.com/squidfunk/mkdocs-material/assets/38702/c4b3f0e6-eb2b-48ab-8e05-13f5de3d5062)
Repo https://github.com/brabster/tw-site-mkdocs

**Comment by squidfunk** [comment]
Thanks for reporting. This isn't specifically related to Material for MkDocs, as it will also happen with other themes, so it's a good idea to report this upstream to the authors of the [RSS plugin](https://github.com/guts/mkdocs-rss-plugin/). My guess is that URLs should be resolved by the plugin to absolute URLs, so RSS feeders render them consistently.

**Comment by alexvoss** [comment]
I have been wondering about how the RSS plugin deals with rendering content as well recently. This is the [place where the Markdown is rendered.](https://github.com/Guts/mkdocs-rss-plugin/blob/77250f01524016b9989623db3c9099f8c5a7bdb4/mkdocs_rss_plugin/util.py#L494) for the RSS feed and it does not give any special treatment to links or anything else. I think that explains the observed behavior. The consequence is to work around this by ensuring that only standard Markdown and HTML links are used in the excerpt of a post.

My use case was that I have an admonition in the excerpt, which does what I want in the blog but in the RSS feed I get the raw Markdown because the admonition syntax is an [extension](https://facelessuser.github.io/pymdown-extensions/extensions/blocks/plugins/admonition/). What I would have liked would have been to get the corresponding HTML `div`s instead. I do wonder what the rationale was to use the raw Markdown as the source instead of the finished HTML?

**Reply by alexvoss** [reply]
As @squidfunk say, best to report this upstream.

**Comment by brabster** [comment]
Raised https://github.com/Guts/mkdocs-rss-plugin/issues/263

**Comment by brabster** [comment]
Hmm I really wanted this fixed so I had a go. Couldn't find a neat solution, so I ended up working directly with the html output but it works for my case.

https://github.com/Guts/mkdocs-rss-plugin/pull/265

https://tempered.works/feed_rss_created.xml

![image](https://github.com/squidfunk/mkdocs-material/assets/38702/674e6b33-355a-452d-8e2b-6da9bb073c63)

I've only implemented just what I need, not sure if @Guts will be interested in doing anything with it, but here's the dependency if anyone wants to have a play (normal use at your own risk disclaimers apply)

`git+https://github.com/brabster/mkdocs-rss-plugin.git@hacky-fix-relative-links-in-feed`

**Comment by brabster** [comment]
I've [updated the PR](https://github.com/Guts/mkdocs-rss-plugin/issues/263#issuecomment-2467782123) over on the RSS plugin repo - I'm not sure there is a neat solution upstream on the RSS plugin, but I might have missed something. I'm sure any thoughts would be helpful!

**Reply by squidfunk** [reply]
It must be fixed in the RSS plugin – let's hope it will be!
